src/lib.rs

```diff
@@ -47,28 +102,106 @@ where
     strfmt_map(fmtstr, &formatter)
 }
 
+macro_rules! display_str_impl {
+    ($($t:ident)*) => ($(
+        impl DisplayStr for $t {
+            fn display_str(&self,f:&mut Formatter) -> Result<()> {
+                f.$t(*self)
+            }
+        }
+    )*)
+}
+
+display_str_impl!(u8 i8 u16 i16 u32 i32 u64 i64 usize isize);
+display_str_impl!(f32 f64);
+
+impl DisplayStr for String{
+    fn display_str(&self, f: &mut Formatter) -> Result<()> {
+        f.str(self.as_str())
+    }
+}
+
+impl DisplayStr for &str{
+    fn display_str(&self, f: &mut Formatter) -> Result<()> {
+        f.str(self)
+    }
+}
+/// This trait is effectively an re-implementation for [std::fmt::Display]
+/// It is used to disguise between the value types that should be formatted
+pub trait DisplayStr {
+    fn display_str(&self, f:&mut Formatter) -> Result<()>;
+}
+
+/// This trait is a shortcut for [strfmt]
+/// for an example see [Format::format]
 pub trait Format {
-    fn format<K, D: fmt::Display>(&self, vars: &HashMap<K, D>) -> Result<String>
-    where
-        K: Hash + Eq + FromStr;
+    /// format a string using strfmt
+    /// # Arguments
+    ///
+    /// * `vars` - A `HashMap` holding the variables to use
+    ///
+    /// # Errors
+    /// Errors are passed directly from strfmt, for details see [strfmt]
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::collections::HashMap;
+    /// use std::f64::consts::PI;
+    /// use strfmt::Format;
+    ///
+    /// let mut my_vars: HashMap<String, f64> = HashMap::new();
+    /// my_vars.insert("Alpha".to_string(),42.0);
+    /// my_vars.insert("Beta".to_string(),PI);
+    ///
+    /// let result = "|{Alpha}|{Beta:<5.2}|".format(&my_vars);
```

Comment:
for examples just use `?`.  So this would be `println!("|{Alpha}|{Beta:<5.2}|".format(&my_vars)?)`

Reply:
oh, sure that's much better